exekutir.xn

```diff
@@ -0,0 +1,140 @@
+---
+# vim-syntax: yaml
+
+# Initial workspace configuration
+- command:
+    contexts:
+        - setup
+    filepath: "/bin/bash"
+    # N/B: Don't use single or double-quotes w/in comments of quoted
+    # shell commands (below).  It will cause a very bad day when things
+    # break in very strange / unexpected ways.  You've been warned!
+    arguments: |
+        -c 'set -e;
+            # Dont modify source files, transfer to workspace
+            echo "Copying ${ADEPT_PATH}/__exekutir__/ -> $WORKSPACE";
+            cp --target-directory="$WORKSPACE" \
+               --recursive \
+               "${ADEPT_PATH}/__exekutir__/"*;
+            echo "-----";
+            # Older ansible doesnt support host/group vars in inventory directory.
+            # Because the ADEPT playbooks treat them the same, just use symlinks.
+            echo "Symlinking ${WORKSPACE}/inventory/{host,group}_vars to $WORKSPACE";
+            ln -sf \
+                "${WORKSPACE}/inventory/host_vars" \
+                "${WORKSPACE}/host_vars";
+            ln -sf \
+                "${WORKSPACE}/inventory/group_vars" \
+                "${WORKSPACE}/group_vars";
+            # Ansible requires ssh keys be setup ahead of time.  Create them if they
+            # dont already exist, otherwise copy them to a standard place
+            # from the $ANSIBLE_PRIVATE_KEY_FILE variable if it was exported.
+            echo "-----";
+            echo "Creating $WORKSPACE/ssh";
+            mkdir -p "$WORKSPACE/ssh";
+            ln -sf "$WORKSPACE/ssh" "$WORKSPACE/.ssh"
+            echo "-----";
+            if [ -r "${ANSIBLE_PRIVATE_KEY_FILE}" ];
+            then
+                echo "Copying ssh keys from $(dirname $ANSIBLE_PRIVATE_KEY_FILE)";
+                cp -a "${ANSIBLE_PRIVATE_KEY_FILE}" "$WORKSPACE/ssh/id_rsa";
```

Comment:
I'm SUPER-uncomfortable with the string 'rsa' in the name. Maybe `id_kommandir` might be better: it doesn't misleadingly imply rsa. And it should be refactored into an envariable, like MY_KEY_FILE because there's a lot of duplication.